mod/blog/start.php

```diff
@@ -278,3 +281,39 @@ function blog_run_upgrades($event, $type, $details) {
 		elgg_set_plugin_setting('upgrade_version', 1, 'blogs');
 	}
 }
+
+/**
+ * Allow liking of a blog
+ *
+ * @param string $hook   "permissions_check:annotate"
+ * @param string $type   "object"
+ * @param array  $return Current value
+ * @param array  $params Hook parameters
+ *
+ * @return void|false
+ */
+function blog_permissions_check_annotate($hook, $type, $return, $params) {
+	if ($return !== false) {
+		// we only want to change to true if it is false
+		return;
+	}
+	
+	if (elgg_extract('annotation_name', $params) !== 'likes') {
+		// we are only interested in liking
+		return;
+	}
+	
+	$user = elgg_extract('user', $params);
+	if (!elgg_instanceof($user, 'user')) {
+		// only users are allowed to annotate
+		return;
+	}
+	
+	$entity = elgg_extract('entity', $params);
+	if (!elgg_instanceof($entity, 'object', 'blog')) {
+		// we are only interested in blogs
+		return;
+	}
```

Comment:
Just grouping the elgg_extract uses might clean this up enough. Then you can make one conditional more cleanly.